src/OrbitClientModel/CaptureSerializer.cpp

```diff
@@ -92,19 +83,35 @@ CaptureInfo GenerateCaptureInfo(
   process->set_command_line(capture_data.process()->command_line());
   process->set_is_64_bit(capture_data.process()->is_64_bit());
 
+  absl::flat_hash_map<std::string, std::string> module_path_to_build_id;
   for (const auto& [module_path, memory_space] : capture_data.process()->GetMemoryMap()) {
-    const ModuleData* module = capture_data.GetModuleByPath(module_path);
+    module_path_to_build_id.insert_or_assign(module_path, memory_space.build_id());
+    const ModuleData* module =
+        capture_data.GetModuleByPathAndBuildId(module_path, memory_space.build_id());
     CHECK(module != nullptr);
     ModuleInfo* module_info = capture_info.add_modules();
     module_info->set_name(module->name());
     module_info->set_file_path(module->file_path());
     module_info->set_file_size(module->file_size());
-    module_info->set_address_start(memory_space.start);
-    module_info->set_address_end(memory_space.end);
+    module_info->set_address_start(memory_space.start());
+    module_info->set_address_end(memory_space.end());
     module_info->set_build_id(module->build_id());
     module_info->set_load_bias(module->load_bias());
   }
 
+  for (const auto& pair : capture_data.instrumented_functions()) {
+    const std::string& module_path = pair.second.file_path();
+    CHECK(module_path_to_build_id.contains(module_path));
```

Comment:
I might miss something, but shouldn't an `InstrumentedFunction` (`pair.second`) now contain a file_build_id? This would make the `module_path_to_build_id` map unnecessary

Reply:
right. thanks for catching this - fixed.